apps/system/test/unit/nfc_handover_manager_test.js

```diff
@@ -93,14 +86,272 @@ suite('Nfc Handover Manager Functions', function() {
+        assert.isFalse(nfcHandoverManager.bluetoothStatusSaved);
+        assert.ok(Service.query
+          .calledWith('BluetoothTransfer.isSendFileQueueEmpty'));
+        assert.ok(window.dispatchEvent.called);
+    });
+
+    test('if handover is not in progress, bluetooth transfer queue is empty, '+
+      'and bluetoothAutoEnabled is false', function() {
+        this.sinon.stub(nfcHandoverManager, 'isHandoverInProgress')
+          .returns(false);
+        this.sinon.stub(Service, 'query').returns(true);
+        this.sinon.spy(window, 'dispatchEvent');
+        nfcHandoverManager.bluetoothAutoEnabled = false;
+        nfcHandoverManager._restoreBluetoothStatus();
+
+        assert.isFalse(nfcHandoverManager.bluetoothAutoEnabled);
+        assert.ok(Service.query
+          .calledWith('BluetoothTransfer.isSendFileQueueEmpty'));
+        assert.ok(!window.dispatchEvent.called);
+    });
+
+    test('if handover is not in progress, bluetooth transfer queue ' +
+      'is not empty', function() {
+        this.sinon.stub(nfcHandoverManager, 'isHandoverInProgress')
+          .returns(false);
+        this.sinon.stub(Service, 'query').returns(false);
+        this.sinon.spy(window, 'dispatchEvent');
+        nfcHandoverManager._restoreBluetoothStatus();
+
+        assert.ok(Service.query
+          .calledWith('BluetoothTransfer.isSendFileQueueEmpty'));
+        assert.ok(!window.dispatchEvent.called);
+    });
+
+    test('if handover is in progress, bluetooth transfer queue is empty',
+      function() {
+        this.sinon.stub(nfcHandoverManager, 'isHandoverInProgress')
+          .returns(true);
+        this.sinon.stub(Service, 'query').returns(true);
+        this.sinon.spy(window, 'dispatchEvent');
+        nfcHandoverManager._restoreBluetoothStatus();
+
+        assert.ok(!Service.query
+          .calledWith('BluetoothTransfer.isSendFileQueueEmpty'));
+        assert.ok(!window.dispatchEvent.called);
+    });
+
+    test('if handover is in progress, bluetooth transfer queue is not empty',
+      function() {
+        this.sinon.stub(nfcHandoverManager, 'isHandoverInProgress')
+          .returns(true);
+        this.sinon.stub(Service, 'query').returns(false);
+        this.sinon.spy(window, 'dispatchEvent');
+        nfcHandoverManager._restoreBluetoothStatus();
+
+        assert.ok(!Service.query
+          .calledWith('BluetoothTransfer.isSendFileQueueEmpty'));
+        assert.ok(!window.dispatchEvent.called);
+    });
+  });
+
+  suite('findPairedDevice', function() {
+    var device = {
+      address: '01:23:45:67:89:AB',
+      connected: true
+    };
+
+    setup(function() {
+      nfcHandoverManager._adapter = MockBTAdapter;
+    });
+
+    teardown(function() {
+      nfcHandoverManager._adapter = null;
+    });
+
+    test('get Bluetooth devices from promise', function() {
+      this.sinon.stub(Service, 'request', function() {
+        return Promise.resolve([device]);
```

Comment:
Please use MockPromise instead of regular Promise (not only here but in other tests also). You need to add /shared/test/unit/mocks/mock_promise.js. Here's an example test using it in NfcManager: https://github.com/mozilla-b2g/gaia/blob/master/apps/system/test/unit/nfc_manager_test.js#L971-L986